`Development`: Add git commit to navbar version in dev mode

TheZoker opened the pull request:
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->

### Description
<!-- Describe your changes in detail -->
This adds a git commit id into the header, when Artemis is run in development mode. 

![image](https://user-images.githubusercontent.com/1368405/184558688-6f3393a6-762f-4e34-afd0-4a4a7ccb411b.png)

Not displayed in the prod mode:
![image](https://user-images.githubusercontent.com/1368405/184558964-dca7deb8-5c34-44d8-bf57-36c682bf346d.png)



### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
1. Start Artemis in dev mode
2. Have a look at the version in the header

### Review Progress
<!-- Each Pull Request should be reviewed by at least two other developers. The code as well as the functionality (= manual test) needs to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
<!-- Please add the test coverages for all changed files here. You can see this when executing the tests locally (see build.gradle and package.json) or when looking into the corresponding Bamboo build plan. -->
<!-- Lines are the main reference but a significantly lower branch percentage can indicate missing edge cases in the tests. -->
| Class/File | Branch | Line |
|------------|-------:|-----:|
<!--
| ExerciseService.java | 85% | 77% |
| programming-exercise.component.ts | 13% | 95% |
-->

### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->

ge65cer reviewed (commented):
I don't think you need a new service class for this. You can probably integrate it into the existing `ProfileService` which is already shared between components.
Or is there a specific reason you implemented it this way?

bassner reviewed (changes requested):
I think the same as @ge65cer. We should not introduce a new service for this. This will duplicate the REST call to the management info endpoint (one from the profile service, one from the new one). Just add the commit and branch as new properties in the profile service

Hialus reviewed (changes requested):
In addition to what Moritz and Patrick said:
I personally don't think it makes much sense just to include the commit hash here. I think showing the branch from which this release was created as well would make more sense. In order to not clutter it you might show one of the values only when hovering over the version.
It's just an idea though, as I think this might be useful to know, as currently the only way to find out what branch is currently deployed is to check it on Bamboo.

ge65cer reviewed (changes requested):
See my comment above.

TheZoker commented:
I removed the additional service and moved to the already existing profile service. Thanks for the hints!

Since [this PR](https://github.com/ls1intum/Artemis/pull/5580), you can see which branch was used by just visiting the `/managment/info` page and it will tell you a lot of information. I think having the git commit in the header when in dev (as suggested by Stefan) is a good idea and if you want to have more information about git, you can look it up in the mentioned endpoint. And [this PR](https://github.com/ls1intum/Artemis/pull/5607) also added the branch name to the startup log.

ge65cer reviewed (commented):
Thanks for the refactor, a lot cleaner now! I tested on TS2 and the endpoint returns the branch name correctly.

It does not display the commit id in the navbar on our test servers, though. (I suppose they run in production mode.) In my opinion, that would be very useful. At best, I could hover over the version and a tooltip would show me commit id and branch name.

Hialus reviewed (dismissed):
Looks good to me. Nice feature.

krusche commented:
I would display the branch name immediately on dev and test systems.
The commit id should be shown in a tooltip on such systems.

On production systems, we should **not** show the commit as tooltip!

Instead, I suggest to show the branch and commit id in the bottom of the about page.

krusche commented:
I would display it directly right to the version number

TheZoker commented:
The branch name in the navbar:
![image](https://user-images.githubusercontent.com/1368405/188154865-a831a670-df26-464f-ab50-d647b268dfc3.png)

On mobile devices and on production system, the branch name will be hidden.

And the about page:
![image](https://user-images.githubusercontent.com/1368405/188154980-367fee1c-6efe-48a8-b803-6d4a94724ae1.png)

stefanwaldhauser reviewed (approved):
Great job! Works perfectly. Tested it locally in dev mode and on the testserver in production mode!

ge65cer commented on src/main/webapp/app/core/about-us/about-us.component.html:
```diff
@@ -56,4 +56,8 @@ <h3 class="font-weight-bold">{{ 'artemisApp.aboutUsOverview.contributors' | arte
     </div>
+    <div class="mt-5 ms-2 me-2">
+        <h3 class="font-weight-bold">Git Information</h3>
+        <div>Git Branch: {{ gitBranchName }}<br />Git Commit ID: {{ gitCommitId }}</div>
```
Even though the translations might be identical here, please move all text to the translation files.

TheZoker replied:
Done :)

ge65cer reviewed (changes requested):
Looks good! Translations & client tests for the about us page need to be adjusted.
As the space taken by the version is now significantly longer in the navbar, this might mess up the responsive layout there and the breakpoints may need to be adjusted. CC @bassner

ge65cer reviewed (approved):
Looks good from my side now. But as I said please wait for the okay from @bassner regarding the navbar.

bassner commented:
Well, yes, this is messing up the breakpoints; if you're adding stuff to the navbar that consumes width, the calculation for the breakpoints needs to be adjusted [here](https://github.com/ls1intum/Artemis/blob/be91820c4fd7d21ed20f4498ec4af4fe87ce2d8f/src/main/webapp/app/shared/layouts/navbar/navbar.component.ts#L146L173).

> At best, I could hover over the version and a tooltip would show me commit id and branch name.

This would probably be the simplest solution for that problem

https://user-images.githubusercontent.com/23171488/192281821-38cf5bfd-f56d-494f-a0bc-e239492dd69d.mov

TheZoker commented:
Thanks for the hint!
The hover style was actually the first approach. 

@krusche
Should I revert it back to hover or do you want to display the information right away?

krusche commented:
> Thanks for the hint! The hover style was actually the first approach.
> 
> @krusche Should I revert it back to hover or do you want to display the information right away?

If it is too complicated to show it in the navbar directly, then I suggest to add it to the bottom of the page (branch + git commit) above or below the following footer
<img width="807" alt="image" src="https://user-images.githubusercontent.com/744067/192379995-df6bc7ca-774e-4d58-835c-4b3ad2b608ec.png">

Of course, this should only be shown for dev and test environments

If you want to show it additionally when hovering over the version, this would be fine for me, but I'd like to see it quickly on dev and test environments without the need to move the mouse ;-)

krusche commented:
I would change `Git Commit ID: a8482b6` to simply `Git commit: a8482b6`
